Modules/System/Barcode Encoder/src/encoders/Codabar/Codabar Encoder.Codeunit.al

```diff
@@ -0,0 +1,182 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved. 
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+/// <summary> 
+/// codabar barcode font implementation from IDAutomation
+/// from: https://www.idautomation.com/barcode-fonts/codabar/ 
+/// A numeric barcode encoding numbers with a slightly higher density than Code 39.
+/// </summary>
+codeunit 9212 CodabarBarcodeEncoder implements IBarcodeEncoder
+{
+    Access = Public;
```

Comment:
The encoders should not be public. They don't need a facade. We should hide the implementation details of those and achieve extensibility through the interfaces. Let's discuss.